AK/LexicalPath.h

```diff
@@ -10,6 +10,11 @@
 #include <AK/String.h>
 #include <AK/Vector.h>
 
+// On Linux distros that use mlibc `basename` is defined as a macro that expands to `__mlibc_gnu_basename` or `__mlibc_gnu_basename_c`, so we undefine it.
+#if defined(__linux__) && defined(basename)
```

Comment:
Ah, I pushed up a PR a few days ago to replace all the `__platform__` macros with `AK_OS_PLATFORM` equivalents for code clarity, could you rebase these to use that pattern? I think then we'll be good to merge em in. I removed the `LibRegex/C/` files from the Lagom build at the same time, so that's one less patch you'll have to carry in your port.

Reply:
Sorry for the long wait, but I've addressed this now. Should be good to go then,